x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/routes/delete.ts

```diff
@@ -76,3 +80,65 @@ export const deleteEntityEngineRoute = (
       }
     );
 };
+
+export const deleteEntityEnginesRoute = (
+  router: EntityAnalyticsRoutesDeps['router'],
+  logger: Logger,
+  getStartServices: EntityAnalyticsRoutesDeps['getStartServices']
+) => {
+  router.versioned
+    .delete({
+      access: 'public',
+      path: '/api/entity_store/engines',
+      security: {
+        authz: {
+          requiredPrivileges: ['securitySolution', `${APP_ID}-entity-analytics`],
+        },
+      },
+    })
+    .addVersion(
+      {
+        version: API_VERSIONS.public.v1,
+        validate: {
+          request: {
+            query: buildRouteValidationWithZod(DeleteEntityEnginesRequestQuery),
+          },
+        },
+      },
+
+      async (context, request, response): Promise<IKibanaResponse<DeleteEntityEnginesResponse>> => {
+        const siemResponse = buildSiemResponse(response);
+        const [_, { taskManager }] = await getStartServices();
+        if (!taskManager) {
+          return siemResponse.error({
+            statusCode: 400,
+            body: TASK_MANAGER_UNAVAILABLE_ERROR,
+          });
+        }
+        try {
+          const secSol = await context.securitySolution;
+          let engines = request.query.entityTypes;
+          if (engines === undefined || engines.length === 0) {
+            engines = [EntityType.user, EntityType.host, EntityType.service, EntityType.generic];
```

Comment:
What happens if one of the engines is disabled?
Should we verify which engines are active before constructing the engines list?